tests/ignite/contrib/metrics/test_cohen_kappa.py

```diff
@@ -0,0 +1,148 @@
+    np.random.seed(1)
+    size = 100
+    np_y_pred = np.random.randint(0, 2, size=(size, 1), dtype=np.long)
+    np_y = np.zeros((size,), dtype=np.long)
+    np_y[size // 2 :] = 1
+    np.random.shuffle(np_y)
+
+    ck_value_sk = cohen_kappa_score(np_y, np_y_pred)
+
+    batch_size = 10
+
+    def update_fn(engine, batch):
+        idx = (engine.state.iteration - 1) * batch_size
+        y_true_batch = np_y[idx : idx + batch_size]
+        y_pred_batch = np_y_pred[idx : idx + batch_size]
+        return idx, torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+
+    engine = Engine(update_fn)
+
+    ck_metric = CohenKappa(output_transform=lambda x: (x[1], x[2]))
+    ck_metric.attach(engine, "cohen_kappa")
+
+    data = list(range(size // batch_size))
+    ck_value = engine.run(data, max_epochs=1).metrics["cohen_kappa"]
+
+    assert np.array_equal(ck_value, ck_value_sk)
+
+
+def test_integration_cohen_kappa_linear_weighted_with_output_transform():
+    np.random.seed(1)
+    size = 100
+    np_y_pred = np.random.randint(0, 2, size=(size, 1), dtype=np.long)
+    np_y = np.zeros((size,), dtype=np.long)
+    np_y[size // 2 :] = 1
+    np.random.shuffle(np_y)
+
+    ck_value_sk = cohen_kappa_score(np_y, np_y_pred, weights="linear")
+
+    batch_size = 10
+
+    def update_fn(engine, batch):
+        idx = (engine.state.iteration - 1) * batch_size
+        y_true_batch = np_y[idx : idx + batch_size]
+        y_pred_batch = np_y_pred[idx : idx + batch_size]
+        return idx, torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+
+    engine = Engine(update_fn)
+
+    ck_metric = CohenKappa(output_transform=lambda x: (x[1], x[2]), weights="linear")
+    ck_metric.attach(engine, "cohen_kappa")
+
+    data = list(range(size // batch_size))
+    ck_value = engine.run(data, max_epochs=1).metrics["cohen_kappa"]
+
+    assert np.array_equal(ck_value, ck_value_sk)
+
+
+def test_integration_cohen_kappa_quadratic_weighted_with_output_transform():
+    np.random.seed(1)
+    size = 100
+    np_y_pred = np.random.randint(0, 2, size=(size, 1), dtype=np.long)
+    np_y = np.zeros((size,), dtype=np.long)
+    np_y[size // 2 :] = 1
+    np.random.shuffle(np_y)
+
+    ck_value_sk = cohen_kappa_score(np_y, np_y_pred, weights="quadratic")
+
+    batch_size = 10
+
+    def update_fn(engine, batch):
+        idx = (engine.state.iteration - 1) * batch_size
+        y_true_batch = np_y[idx : idx + batch_size]
+        y_pred_batch = np_y_pred[idx : idx + batch_size]
+        return idx, torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+
+    engine = Engine(update_fn)
+
+    ck_metric = CohenKappa(output_transform=lambda x: (x[1], x[2]), weights="quadratic")
+    ck_metric.attach(engine, "cohen_kappa")
+
+    data = list(range(size // batch_size))
+    ck_value = engine.run(data, max_epochs=1).metrics["cohen_kappa"]
+
+    assert np.array_equal(ck_value, ck_value_sk)
```

Comment:
We need to add distributed tests as well. See https://github.com/pytorch/ignite/blob/master/tests/ignite/contrib/metrics/regression/test_canberra_metric.py#L99

Reply:
We are using sklearn just like average precision, can we change the device? I guess it's not possible, the same thing for `roc curve`, `average precision`, `precision recall curve`.

Reply:
Yes, you are right, we are missing an argument in the constructor, see https://github.com/pytorch/ignite/blob/d38c8bcb1794b85c3525c8cc1862f7e520d7862c/ignite/metrics/epoch_metric.py#L44

The point here is that we have to start providing distributed support for all metrics (see https://github.com/pytorch/ignite/issues/1284). EpochMetric, base class for this metric, now supports distributed configuration and performs necessary collective op (all_gather) to fetch all the data from each participating proc and executes `compute_fn` on the rank 0 passing gathered data (https://github.com/pytorch/ignite/blob/d38c8bcb1794b85c3525c8cc1862f7e520d7862c/ignite/metrics/epoch_metric.py#L123-L147).
Currently, we are assuming that device is CPU and gathered tensors are also on CPU. And yes, you are correct, that if user specifies `device` as CUDA, calling `y_targets.numpy()` will fail. We have to fix that and as I said add distributed tests on cpu and gpu.

Reply:
Okay i will add the device arg for now and check if the user specified CUDA then i will raise an exception that it's not supported yet.

Reply:
You can simply call `y_targets.cpu().numpy()` for other device than cpu. No need to raise an error.